types/canonical.go

```diff
@@ -51,26 +51,30 @@ func CanonicalizeProposal(chainID string, proposal *tmproto.Proposal) tmproto.Ca
 	}
 }
 
-func GetVoteExtensionToSign(ext *tmproto.VoteExtension) *tmproto.VoteExtensionToSign {
-	if ext == nil {
-		return nil
-	}
-	return &tmproto.VoteExtensionToSign{
-		AppDataToSign: ext.AppDataToSign,
-	}
-}
-
 // CanonicalizeVote transforms the given Vote to a CanonicalVote, which does
-// not contain ValidatorIndex and ValidatorAddress fields.
+// not contain ValidatorIndex and ValidatorAddress fields, or any fields
+// relating to vote extensions.
 func CanonicalizeVote(chainID string, vote *tmproto.Vote) tmproto.CanonicalVote {
 	return tmproto.CanonicalVote{
-		Type:          vote.Type,
-		Height:        vote.Height,       // encoded as sfixed64
-		Round:         int64(vote.Round), // encoded as sfixed64
-		BlockID:       CanonicalizeBlockID(vote.BlockID),
-		Timestamp:     vote.Timestamp,
-		ChainID:       chainID,
-		VoteExtension: GetVoteExtensionToSign(vote.VoteExtension),
+		Type:      vote.Type,
+		Height:    vote.Height,       // encoded as sfixed64
+		Round:     int64(vote.Round), // encoded as sfixed64
+		BlockID:   CanonicalizeBlockID(vote.BlockID),
+		Timestamp: vote.Timestamp,
+		ChainID:   chainID,
+	}
+}
+
+// CanonicalizeVoteExtension extracts the vote extension from the given vote
+// and constructs a CanonicalizeVoteExtension struct, whose representation in
+// bytes is what is signed in order to produce the vote extension's signature.
+func CanonicalizeVoteExtension(chainID string, vote *tmproto.Vote) tmproto.CanonicalVoteExtension {
```

Comment:
I don't understand why this is called "Cannonicalizing". `Commit`s can get cannonicalized by the chain because there can be several valid representations of a commit which get "cannonicalized" to a single agreed upon representation. But here we're just extracting out a few fields to form the bytes that need to get signed over. Correct me if I'm wrong but this doesn't sound like "Cannonicalizing". Maybe use something like "plaintext", "preimage" or "digest" to signal that this is the data that gets signed over.

Reply:
I agree "canonical" and "canonicalize" are being used for two different meanings in Tendermint:
* To name the commit for height _h_ (including the round) that the network agrees to store as `LastCommit` value in _h+1_, among all the possible commits for _h_
* To name the construction made in memory to sign votes where some information (chainId, height, round) is added at the moment of signing, but then is removed from the final vote message because it can be derived by other means at the moment of checking the signature (and would then be redundant)

Bear in mind that this _double meaning_ predates vote extensions. So the name `CanonicalizeVoteExtension` here is just consistent with `CanonicalizeVote` or `CanonicalizeProposal` in the same file.

Having said that, I totally agree that, moving forward, we should probably decide on one meaning of "canonical" and drop the other. It will help in many discussions.

Reply:
#8332

Reply:
Since this is purely an internal construction, we could simply just remove the `CanonicalVoteExtension` proto and type altogether. They're only used in one place anyways: https://github.com/tendermint/tendermint/blob/8bb7983c4c3f0956329539cf3b4951b389a42e26/types/vote.go#L111-L119

Are we going to need this `CanonicalVoteExtension` (or whatever we call it) anywhere else in future?

Reply:
Thanks Thane!